src/Form/AddContext.php

```diff
@@ -0,0 +1,204 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\pathauto\Form\AddContext.
+ */
+
+namespace Drupal\pathauto\Form;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\OpenModalDialogCommand;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormBuilderInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+class AddContext extends FormBase {
+
+  /**
+   * @var string
+   */
+  protected $machine_name;
+
+  /**
+   * An array of property types that are eligible as relationships.
+   *
+   * @var array
+   */
+  protected $property_types = ['entity_reference', 'language'];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'ctools_add_context_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $cached_values = $form_state->getTemporaryValue('wizard');
+    $this->machine_name = $cached_values['id'];
+    $form['items'] = array(
+      '#type' => 'markup',
+      '#prefix' => '<div id="configured-contexts">',
+      '#suffix' => '</div>',
```

Comment:
It's so lame that we're preaching for years that prefix/suffix is bad but can't come up with a way to do ajax without it ;)